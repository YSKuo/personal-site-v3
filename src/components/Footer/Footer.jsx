import React from "react";
import UserLinks from "../UserLinks/UserLinks";
import styled from "styled-components";
import { Container, Grid, Typography, Link, Divider } from "@material-ui/core";
import { mediaQueryBreakpoint } from "../../constants/breakpoint";

function Footer({ config }) {
  const { userName, userEmail, userResume, copyright } = config;

  return (
    <StyledContainer component="footer">
      <Grid container justify="space-between">
        <PersonalInfoContainer item xs={12} sm={6}>
          <Typography component="p" variant="h5" color="inherit" gutterBottom>
            {userName}
          </Typography>
          <Link href={`mailto:${userEmail}`}>{userEmail}</Link>
        </PersonalInfoContainer>
        <IconContainer item component="nav" xs={12} sm={6}>
          <UserLinks config={config} labeled />
        </IconContainer>
      </Grid>

      <StyledDivider />

      <Grid container justify="space-between">
        <SiteInfoContainer item>
          <Typography component="p" variant="body2" gutterBottom>
            {copyright}
          </Typography>
          <Typography component="p" variant="body2" color="inherit">
            Powered by <Link href="https://www.gatsbyjs.com/">Gatsby</Link> and{" "}
            <Link href="https://github.com/Vagr9K/gatsby-advanced-starter">
              Gatsby Advanced Starter
            </Link>
            .
          </Typography>
        </SiteInfoContainer>
        <Grid item>
          <Typography component="p" variant="body2">
            Download{" "}
            <Link target="_blank" href={userResume}>
              Resume
            </Link>
          </Typography>
        </Grid>
      </Grid>
    </StyledContainer>
  );
}

const StyledContainer = styled(Container)`
  padding-top: 2rem;
  padding-bottom: 2rem;

  ${mediaQueryBreakpoint("sm")} {
    padding-top: 2.5rem;
    padding-bottom: 2.5rem;
  }

  ${mediaQueryBreakpoint("md")} {
    padding-top: 3rem;
    padding-bottom: 3rem;
  }
`;

const PersonalInfoContainer = styled(Grid)`
  display: flex;
  justify-content: flex-start;
  flex-direction: column;
  margin-bottom: 1rem;

  ${mediaQueryBreakpoint("sm")} {
    justify-content: flex-end;
    margin: 0;
  }
`;

const IconContainer = styled(Grid)`
  display: flex;
  justify-content: flex-start;

  ${mediaQueryBreakpoint("sm")} {
    justify-content: flex-end;
  }
`;

const SiteInfoContainer = styled(Grid)`
  display: flex;
  flex-direction: column;
  margin-bottom: 1rem;
`;

const StyledDivider = styled(Divider)`
  margin: 1rem 0;
`;

export default Footer;
