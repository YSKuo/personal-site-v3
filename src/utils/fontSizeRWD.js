import { fontSizes } from "../constants/font";
import { mediaQueryBreakpoint } from "../constants/breakpoint";

function fontSizeRWD(text, [xs, sm, md]) {
  return `${text} {
    ${mediaQueryBreakpoint("xs")} {
      font-size: ${fontSizes[xs]};
    }
    ${mediaQueryBreakpoint("sm")} {
      font-size: ${fontSizes[sm]};
    }
    ${mediaQueryBreakpoint("md")} {
      font-size: ${fontSizes[md]};
    }
  }`;
}

export default fontSizeRWD;
