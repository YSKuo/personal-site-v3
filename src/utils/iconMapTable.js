import React from "react";

import InstagramIcon from "@material-ui/icons/Instagram";
import LinkedInIcon from "@material-ui/icons/LinkedIn";
import GitHubIcon from "@material-ui/icons/GitHub";
import EmailIcon from "@material-ui/icons/Email";
import TwitterIcon from "@material-ui/icons/Twitter";

const iconMapTable = {
  Email: <EmailIcon />,
  Twitter: <TwitterIcon />,
  GitHub: <GitHubIcon />,
  Instagram: <InstagramIcon />,
  LinkedIn: <LinkedInIcon />,
};

export default iconMapTable;
