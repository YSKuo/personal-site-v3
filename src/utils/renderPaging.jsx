import React from "react";
import styled from "styled-components";
import { Grid, Button, IconButton, Hidden } from "@material-ui/core";
import ArrowBackIosIcon from "@material-ui/icons/ArrowBackIos";
import ArrowForwardIosIcon from "@material-ui/icons/ArrowForwardIos";
import MoreHorizIcon from "@material-ui/icons/MoreHoriz";

function renderPaging(pageName, pageContext, location) {
  const { currentPageNum, pageCount } = pageContext;
  const prevPage =
    currentPageNum - 1 === 1
      ? `/${pageName}`
      : `/${pageName}/${currentPageNum - 1}/`;
  const nextPage = `/${pageName}/${currentPageNum + 1}/`;
  const isFirstPage = currentPageNum === 1;
  const isLastPage = currentPageNum === pageCount;
  const currentPage = location.pathname.match(/[0-9]+/)
    ? Number(location.pathname.match(/[0-9]+/))
    : 1;

  return (
    <Grid container component="nav" alignItems="center">
      <PaginationPreviousButton item xs={6} sm={2}>
        {!isFirstPage && (
          <Button
            href={prevPage}
            color="primary"
            variant="outlined"
            startIcon={<ArrowBackIosIcon />}
          >
            Previous
          </Button>
        )}
      </PaginationPreviousButton>

      <Hidden xsDown>
        <Grid item sm={8}>
          <PaginationList>
            {[...Array(pageCount)].map((_val, index) => {
              const pageNum = index + 1;
              if (
                pageNum === 1 ||
                pageNum === pageCount ||
                pageNum === currentPage ||
                pageNum === currentPage + 1 ||
                pageNum === currentPage - 1
              ) {
                return (
                  <PaginationNumber key={`${pageName}-page-${pageNum}`}>
                    <PaginationLink
                      variant={pageNum === currentPage ? "contained" : "text"}
                      size={pageNum === currentPage ? "medium" : "small"}
                      color={pageNum === currentPage ? "secondary" : "default"}
                      href={
                        pageNum === 1
                          ? `/${pageName}`
                          : `/${pageName}/${pageNum}/`
                      }
                      disableElevation
                    >
                      {pageNum}
                    </PaginationLink>
                  </PaginationNumber>
                );
              }
              if (pageNum === currentPage + 2 || pageNum === currentPage - 2) {
                return (
                  <PaginationNumber key={`${pageName}-page-${pageNum}`}>
                    <IconButton title="ellipsis" disabled>
                      <MoreHorizIcon />
                    </IconButton>
                  </PaginationNumber>
                );
              }
            })}
          </PaginationList>
        </Grid>
      </Hidden>
      <PaginationNextButton item xs={6} sm={2}>
        {!isLastPage && (
          <Button
            href={nextPage}
            color="primary"
            variant="outlined"
            endIcon={<ArrowForwardIosIcon />}
          >
            Next
          </Button>
        )}
      </PaginationNextButton>
    </Grid>
  );
}

export default renderPaging;

const PaginationList = styled.ul`
  display: flex;
  text-align: center;
  justify-content: center;
  align-items: center;
  list-style: none;
`;
const PaginationNumber = styled.li``;
const PaginationLink = styled(Button)`
  height: 2rem;
  width: 2rem;
  border-radius: 50%;
  min-width: initial;
  padding: 0 0.5rem;
  margin: 0 1rem;
`;

const PaginationPreviousButton = styled(Grid)`
  text-align: start;
`;

const PaginationNextButton = styled(Grid)`
  text-align: end;
`;
