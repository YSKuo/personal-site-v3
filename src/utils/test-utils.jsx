import React from "react";
import { ThemeProvider, createGlobalStyle } from "styled-components";
import { createMuiTheme, MuiThemeProvider } from "@material-ui/core/styles";
import { render } from "@testing-library/react";
import { theme } from "../constants/theme";

const SizeWrapper = ({ children }) => {
  const initTheme = createMuiTheme({
    props: { MuiWithWidth: { initialWidth: "lg" } },
  });

  return (
    <MuiThemeProvider theme={initTheme}>
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    </MuiThemeProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: SizeWrapper, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render, SizeWrapper };
