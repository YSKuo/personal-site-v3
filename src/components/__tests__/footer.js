import React from "react";

import { render } from "../../utils/test-utils";
import Footer from "../Footer";
import config from "../../../data/SiteConfig";

function renderFooter(props) {
  const utils = render(<Footer config={config} />);

  const footer = utils.getByRole("contentinfo");
  return { ...utils, footer };
}

test("Footer snapshot at lg screen width", () => {
  const { footer } = renderFooter();
  expect(footer).toMatchSnapshot();
});
