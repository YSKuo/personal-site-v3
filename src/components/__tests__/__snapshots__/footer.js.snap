// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer render 1`] = `
<footer
  className="MuiContainer-root sc-bdfBwQ icruZb MuiContainer-maxWidthLg"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-between"
  >
    <div
      className="MuiGrid-root sc-gsTCUz egUoRK MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
    >
      <p
        className="MuiTypography-root MuiTypography-h5 MuiTypography-colorInherit MuiTypography-gutterBottom"
      >
        Yan-Sheng (Arsene) Kuo
      </p>
      <a
        className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
        href="mailto:crowley3141@hotmail.com"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        crowley3141@hotmail.com
      </a>
    </div>
    <nav
      className="MuiGrid-root sc-dlfnbm dMlzcG MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
    >
      <a
        className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
        href="https://github.com/YSKuo"
        onBlur={[Function]}
        onFocus={[Function]}
        target="_blank"
      >
        <button
          aria-label="GitHub"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 .3a12 12 0 0 0-3.8 23.4c.6.1.8-.3.8-.6v-2c-3.3.7-4-1.6-4-1.6-.6-1.4-1.4-1.8-1.4-1.8-1-.7.1-.7.1-.7 1.2 0 1.9 1.2 1.9 1.2 1 1.8 2.8 1.3 3.5 1 0-.8.4-1.3.7-1.6-2.7-.3-5.5-1.3-5.5-6 0-1.2.5-2.3 1.3-3.1-.2-.4-.6-1.6 0-3.2 0 0 1-.3 3.4 1.2a11.5 11.5 0 0 1 6 0c2.3-1.5 3.3-1.2 3.3-1.2.6 1.6.2 2.8 0 3.2.9.8 1.3 1.9 1.3 3.2 0 4.6-2.8 5.6-5.5 5.9.5.4.9 1 .9 2.2v3.3c0 .3.1.7.8.6A12 12 0 0 0 12 .3"
              />
            </svg>
          </span>
        </button>
      </a>
      <a
        className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
        href="https://twitter.com/ArseneKuo"
        onBlur={[Function]}
        onFocus={[Function]}
        target="_blank"
      >
        <button
          aria-label="Twitter"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22.46 6c-.77.35-1.6.58-2.46.69.88-.53 1.56-1.37 1.88-2.38-.83.5-1.75.85-2.72 1.05C18.37 4.5 17.26 4 16 4c-2.35 0-4.27 1.92-4.27 4.29 0 .34.04.67.11.98C8.28 9.09 5.11 7.38 3 4.79c-.37.63-.58 1.37-.58 2.15 0 1.49.75 2.81 1.91 3.56-.71 0-1.37-.2-1.95-.5v.03c0 2.08 1.48 3.82 3.44 4.21a4.22 4.22 0 0 1-1.93.07 4.28 4.28 0 0 0 4 2.98 8.521 8.521 0 0 1-5.33 1.84c-.34 0-.68-.02-1.02-.06C3.44 20.29 5.7 21 8.12 21 16 21 20.33 14.46 20.33 8.79c0-.19 0-.37-.01-.56.84-.6 1.56-1.36 2.14-2.23z"
              />
            </svg>
          </span>
        </button>
      </a>
      <a
        className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
        href="https://www.instagram.com/arsene_alibi/"
        onBlur={[Function]}
        onFocus={[Function]}
        target="_blank"
      >
        <button
          aria-label="Instagram"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7.8 2h8.4C19.4 2 22 4.6 22 7.8v8.4a5.8 5.8 0 0 1-5.8 5.8H7.8C4.6 22 2 19.4 2 16.2V7.8A5.8 5.8 0 0 1 7.8 2m-.2 2A3.6 3.6 0 0 0 4 7.6v8.8C4 18.39 5.61 20 7.6 20h8.8a3.6 3.6 0 0 0 3.6-3.6V7.6C20 5.61 18.39 4 16.4 4H7.6m9.65 1.5a1.25 1.25 0 0 1 1.25 1.25A1.25 1.25 0 0 1 17.25 8 1.25 1.25 0 0 1 16 6.75a1.25 1.25 0 0 1 1.25-1.25M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3z"
              />
            </svg>
          </span>
        </button>
      </a>
      <a
        className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
        href="https://www.linkedin.com/in/yanshengkuo/"
        onBlur={[Function]}
        onFocus={[Function]}
        target="_blank"
      >
        <button
          aria-label="LinkedIn"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
              />
            </svg>
          </span>
        </button>
      </a>
    </nav>
  </div>
  <hr
    className="MuiDivider-root sc-eCssSg ittpIq"
  />
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-between"
  >
    <div
      className="MuiGrid-root sc-hKgILt jOJUsX MuiGrid-item"
    >
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom"
      >
        Copyright Â© Arsene Kuo 2019 - 2021
      </p>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-colorInherit"
      >
        Powered by
         
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
          href="https://www.gatsbyjs.com/"
          onBlur={[Function]}
          onFocus={[Function]}
          target="_blank"
        >
          Gatsby
        </a>
         
        and
         
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
          href="https://github.com/Vagr9K/gatsby-advanced-starter"
          onBlur={[Function]}
          onFocus={[Function]}
          target="_blank"
        >
          Gatsby Advanced Starter
        </a>
        .
      </p>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <p
        className="MuiTypography-root MuiTypography-body2"
      >
        Download
         
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
          href="https://drive.google.com/file/d/1dhQxH2l2zC5rxhzzmzvPZHOyFTHfOxW0/view?usp=sharing"
          onBlur={[Function]}
          onFocus={[Function]}
          target="_blank"
        >
          Resume
        </a>
      </p>
    </div>
  </div>
</footer>
`;

exports[`Footer snapshot render 1`] = `
<footer
  className="MuiContainer-root sc-bdfBwQ icruZb MuiContainer-maxWidthLg"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-between"
  >
    <div
      className="MuiGrid-root sc-gsTCUz egUoRK MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
    >
      <p
        className="MuiTypography-root MuiTypography-h5 MuiTypography-colorInherit MuiTypography-gutterBottom"
      >
        Yan-Sheng (Arsene) Kuo
      </p>
      <a
        className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
        href="mailto:crowley3141@hotmail.com"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        crowley3141@hotmail.com
      </a>
    </div>
    <nav
      className="MuiGrid-root sc-dlfnbm dMlzcG MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
    >
      <a
        className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
        href="https://github.com/YSKuo"
        onBlur={[Function]}
        onFocus={[Function]}
        target="_blank"
      >
        <button
          aria-label="GitHub"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 .3a12 12 0 0 0-3.8 23.4c.6.1.8-.3.8-.6v-2c-3.3.7-4-1.6-4-1.6-.6-1.4-1.4-1.8-1.4-1.8-1-.7.1-.7.1-.7 1.2 0 1.9 1.2 1.9 1.2 1 1.8 2.8 1.3 3.5 1 0-.8.4-1.3.7-1.6-2.7-.3-5.5-1.3-5.5-6 0-1.2.5-2.3 1.3-3.1-.2-.4-.6-1.6 0-3.2 0 0 1-.3 3.4 1.2a11.5 11.5 0 0 1 6 0c2.3-1.5 3.3-1.2 3.3-1.2.6 1.6.2 2.8 0 3.2.9.8 1.3 1.9 1.3 3.2 0 4.6-2.8 5.6-5.5 5.9.5.4.9 1 .9 2.2v3.3c0 .3.1.7.8.6A12 12 0 0 0 12 .3"
              />
            </svg>
          </span>
        </button>
      </a>
      <a
        className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
        href="https://twitter.com/ArseneKuo"
        onBlur={[Function]}
        onFocus={[Function]}
        target="_blank"
      >
        <button
          aria-label="Twitter"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22.46 6c-.77.35-1.6.58-2.46.69.88-.53 1.56-1.37 1.88-2.38-.83.5-1.75.85-2.72 1.05C18.37 4.5 17.26 4 16 4c-2.35 0-4.27 1.92-4.27 4.29 0 .34.04.67.11.98C8.28 9.09 5.11 7.38 3 4.79c-.37.63-.58 1.37-.58 2.15 0 1.49.75 2.81 1.91 3.56-.71 0-1.37-.2-1.95-.5v.03c0 2.08 1.48 3.82 3.44 4.21a4.22 4.22 0 0 1-1.93.07 4.28 4.28 0 0 0 4 2.98 8.521 8.521 0 0 1-5.33 1.84c-.34 0-.68-.02-1.02-.06C3.44 20.29 5.7 21 8.12 21 16 21 20.33 14.46 20.33 8.79c0-.19 0-.37-.01-.56.84-.6 1.56-1.36 2.14-2.23z"
              />
            </svg>
          </span>
        </button>
      </a>
      <a
        className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
        href="https://www.instagram.com/arsene_alibi/"
        onBlur={[Function]}
        onFocus={[Function]}
        target="_blank"
      >
        <button
          aria-label="Instagram"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7.8 2h8.4C19.4 2 22 4.6 22 7.8v8.4a5.8 5.8 0 0 1-5.8 5.8H7.8C4.6 22 2 19.4 2 16.2V7.8A5.8 5.8 0 0 1 7.8 2m-.2 2A3.6 3.6 0 0 0 4 7.6v8.8C4 18.39 5.61 20 7.6 20h8.8a3.6 3.6 0 0 0 3.6-3.6V7.6C20 5.61 18.39 4 16.4 4H7.6m9.65 1.5a1.25 1.25 0 0 1 1.25 1.25A1.25 1.25 0 0 1 17.25 8 1.25 1.25 0 0 1 16 6.75a1.25 1.25 0 0 1 1.25-1.25M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3z"
              />
            </svg>
          </span>
        </button>
      </a>
      <a
        className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
        href="https://www.linkedin.com/in/yanshengkuo/"
        onBlur={[Function]}
        onFocus={[Function]}
        target="_blank"
      >
        <button
          aria-label="LinkedIn"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
              />
            </svg>
          </span>
        </button>
      </a>
    </nav>
  </div>
  <hr
    className="MuiDivider-root sc-eCssSg ittpIq"
  />
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-between"
  >
    <div
      className="MuiGrid-root sc-hKgILt jOJUsX MuiGrid-item"
    >
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom"
      >
        Copyright Â© Arsene Kuo 2019 - 2021
      </p>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-colorInherit"
      >
        Powered by
         
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
          href="https://www.gatsbyjs.com/"
          onBlur={[Function]}
          onFocus={[Function]}
          target="_blank"
        >
          Gatsby
        </a>
         
        and
         
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
          href="https://github.com/Vagr9K/gatsby-advanced-starter"
          onBlur={[Function]}
          onFocus={[Function]}
          target="_blank"
        >
          Gatsby Advanced Starter
        </a>
        .
      </p>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <p
        className="MuiTypography-root MuiTypography-body2"
      >
        Download
         
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
          href="https://drive.google.com/file/d/1OdNg980V-9H4kksJgmXNlRqpd2KLgupI/view?usp=sharing"
          onBlur={[Function]}
          onFocus={[Function]}
          target="_blank"
        >
          Resume
        </a>
      </p>
    </div>
  </div>
</footer>
`;
